Update APIs of nvinfer

### Checklist

- [X] I've read the [contribution guidelines](https://github.com/autowarefoundation/autoware/blob/main/CONTRIBUTING.md).
- [X] I've searched other issues and no duplicate issues were found.
- [X] I've agreed with the maintainers that I can plan this task.

### Description

```
--- stderr: tensorrt_yolo
/home/kenji/autoware/src/universe/autoware.universe/perception/tensorrt_yolo/lib/src/trt_yolo.cpp: In member function ‘void yolo::Net::load(const string&)’:
/home/kenji/autoware/src/universe/autoware.universe/perception/tensorrt_yolo/lib/src/trt_yolo.cpp:69:72: error: ‘nvinfer1::ICudaEngine* nvinfer1::IRuntime::deserializeCudaEngine(const void*, std::size_t, nvinfer1::IPluginFactory*)’ is deprecated [-Werror=deprecated-declarations]
   69 |       unique_ptr<nvinfer1::ICudaEngine>(runtime_->deserializeCudaEngine(buffer, size, nullptr));
      |                                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/x86_64-linux-gnu/NvInfer.h:54,
                 from /home/kenji/autoware/src/universe/autoware.universe/perception/tensorrt_yolo/lib/include/calibrator.hpp:45,
                 from /home/kenji/autoware/src/universe/autoware.universe/perception/tensorrt_yolo/lib/src/trt_yolo.cpp:37:
/usr/include/x86_64-linux-gnu/NvInferRuntime.h:672:43: note: declared here
  672 |     TRT_DEPRECATED nvinfer1::ICudaEngine* deserializeCudaEngine(
      |                                           ^~~~~~~~~~~~~~~~~~~~~
/home/kenji/autoware/src/universe/autoware.universe/perception/tensorrt_yolo/lib/src/trt_yolo.cpp: In constructor ‘yolo::Net::Net(const string&, const string&, int, const yolo::Config&, const std::vector<std::__cxx11::basic_string<char> >&, const string&, bool, size_t)’:
/home/kenji/autoware/src/universe/autoware.universe/perception/tensorrt_yolo/lib/src/trt_yolo.cpp:256:77: error: ‘nvinfer1::ICudaEngine* nvinfer1::IBuilder::buildEngineWithConfig(nvinfer1::INetworkDefinition&, nvinfer1::IBuilderConfig&)’ is deprecated [-Werror=deprecated-declarations]
  256 |   engine_ = unique_ptr<nvinfer1::ICudaEngine>(builder->buildEngineWithConfig(*network, *config));
      |                                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
In file included from /home/kenji/autoware/src/universe/autoware.universe/perception/tensorrt_yolo/lib/include/calibrator.hpp:45,
                 from /home/kenji/autoware/src/universe/autoware.universe/perception/tensorrt_yolo/lib/src/trt_yolo.cpp:37:
/usr/include/x86_64-linux-gnu/NvInfer.h:8640:43: note: declared here
 8640 |     TRT_DEPRECATED nvinfer1::ICudaEngine* buildEngineWithConfig(
      |                                           ^~~~~~~~~~~~~~~~~~~~~
In file included from /home/kenji/autoware/src/universe/autoware.universe/perception/tensorrt_yolo/lib/src/trt_yolo.cpp:41:
/home/kenji/autoware/src/universe/autoware.universe/perception/tensorrt_yolo/lib/include/trt_yolo.hpp: In instantiation of ‘void yolo::Deleter::operator()(T*) const [with T = nvinfer1::IRuntime]’:
/usr/include/c++/11/bits/unique_ptr.h:361:17:   required from ‘std::unique_ptr<_Tp, _Dp>::~unique_ptr() [with _Tp = nvinfer1::IRuntime; _Dp = yolo::Deleter]’
/home/kenji/autoware/src/universe/autoware.universe/perception/tensorrt_yolo/lib/include/trt_yolo.hpp:132:45:   required from here
/home/kenji/autoware/src/universe/autoware.universe/perception/tensorrt_yolo/lib/include/trt_yolo.hpp:62:19: error: ‘void nvinfer1::IRuntime::destroy()’ is deprecated [-Werror=deprecated-declarations]
   62 |       obj->destroy();
      |       ~~~~~~~~~~~~^~
In file included from /usr/include/x86_64-linux-gnu/NvInfer.h:54,
                 from /home/kenji/autoware/src/universe/autoware.universe/perception/tensorrt_yolo/lib/include/calibrator.hpp:45,
                 from /home/kenji/autoware/src/universe/autoware.universe/perception/tensorrt_yolo/lib/src/trt_yolo.cpp:37:
/usr/include/x86_64-linux-gnu/NvInferRuntime.h:716:25: note: declared here
  716 |     TRT_DEPRECATED void destroy() noexcept
      |                         ^~~~~~~
In file included from /home/kenji/autoware/src/universe/autoware.universe/perception/tensorrt_yolo/lib/src/trt_yolo.cpp:41:
/home/kenji/autoware/src/universe/autoware.universe/perception/tensorrt_yolo/lib/include/trt_yolo.hpp: In instantiation of ‘void yolo::Deleter::operator()(T*) const [with T = nvinfer1::ICudaEngine]’:
/usr/include/c++/11/bits/unique_ptr.h:361:17:   required from ‘std::unique_ptr<_Tp, _Dp>::~unique_ptr() [with _Tp = nvinfer1::ICudaEngine; _Dp = yolo::Deleter]’
/home/kenji/autoware/src/universe/autoware.universe/perception/tensorrt_yolo/lib/include/trt_yolo.hpp:133:47:   required from here
/home/kenji/autoware/src/universe/autoware.universe/perception/tensorrt_yolo/lib/include/trt_yolo.hpp:62:19: error: ‘void nvinfer1::ICudaEngine::destroy()’ is deprecated [-Werror=deprecated-declarations]
   62 |       obj->destroy();
      |       ~~~~~~~~~~~~^~
```

### Purpose

To resolve warnings.

### Possible approaches

Use new APIs.

### Definition of done

No error after `-Wno-deprecated-declarations` is removed.

- [ ] tensorrt_yolo
- [ ] traffic_light_ssd_fine_detector
- [ ] traffic_light_classifier
- [ ] lidar_centerpoint
- [ ] lidar_apollo_instance_segmentation

--- a/perception/lidar_apollo_instance_segmentation/CMakeLists.txt
+++ b/perception/lidar_apollo_instance_segmentation/CMakeLists.txt
@@ -4,9 +4,6 @@ project(lidar_apollo_instance_segmentation)
 find_package(autoware_cmake REQUIRED)
 autoware_package()
 
-# TODO: remove (https://github.com/autowarefoundation/autoware.universe/issues/850)
-add_compile_options(-Wno-deprecated-declarations)
-
 option(CUDA_VERBOSE "Verbose output of CUDA modules" OFF)
 
 # set flags for CUDA availability

--- a/perception/lidar_apollo_instance_segmentation/lib/include/TrtNet.hpp
+++ b/perception/lidar_apollo_instance_segmentation/lib/include/TrtNet.hpp
@@ -56,13 +56,13 @@ public:
     }
 
     if (!mTrtRunTime) {
-      mTrtRunTime->destroy();
+      delete mTrtRunTime;
     }
     if (!mTrtContext) {
-      mTrtContext->destroy();
+      delete mTrtContext;
     }
     if (!mTrtEngine) {
-      mTrtEngine->destroy();
+      delete mTrtEngine;
     }
   }
 

--- a/perception/lidar_apollo_instance_segmentation/lib/src/TrtNet.cpp
+++ b/perception/lidar_apollo_instance_segmentation/lib/src/TrtNet.cpp
@@ -100,7 +100,7 @@ trtNet::trtNet(const std::string & engineFile)
 
   mTrtRunTime = createInferRuntime(gLogger);
   assert(mTrtRunTime != nullptr);
-  mTrtEngine = mTrtRunTime->deserializeCudaEngine(data.get(), length, nullptr);
+  mTrtEngine = mTrtRunTime->deserializeCudaEngine(data.get(), length);
   assert(mTrtEngine != nullptr);
 
   InitEngine();

--- a/perception/lidar_apollo_instance_segmentation/src/detector.cpp
+++ b/perception/lidar_apollo_instance_segmentation/src/detector.cpp
@@ -69,19 +69,19 @@ LidarApolloInstanceSegmentation::LidarApolloInstanceSegmentation(rclcpp::Node * 
     assert(!outfile.fail());
     outfile.write(reinterpret_cast<char *>(plan->data()), plan->size());
     outfile.close();
-    if(network) {
+    if (network) {
       delete network;
     }
-    if(parser) {
+    if (parser) {
       delete parser;
     }
-    if(builder) {
+    if (builder) {
       delete builder;
     }
-    if(config) {
+    if (config) {
       delete config;
     }
-    if(plan) {
+    if (plan) {
       delete plan;
     }
   }

--- a/perception/lidar_apollo_instance_segmentation/src/detector.cpp
+++ b/perception/lidar_apollo_instance_segmentation/src/detector.cpp
@@ -63,17 +63,27 @@ LidarApolloInstanceSegmentation::LidarApolloInstanceSegmentation(rclcpp::Node * 
     const int batch_size = 1;
     builder->setMaxBatchSize(batch_size);
     config->setMaxWorkspaceSize(1 << 30);
-    nvinfer1::ICudaEngine * engine = builder->buildEngineWithConfig(*network, *config);
-    nvinfer1::IHostMemory * trt_model_stream = engine->serialize();
-    assert(trt_model_stream != nullptr);
+    nvinfer1::IHostMemory * plan = builder->buildSerializedNetwork(*network, *config);
+    assert(plan != nullptr);
     std::ofstream outfile(engine_file, std::ofstream::binary);
     assert(!outfile.fail());
-    outfile.write(reinterpret_cast<char *>(trt_model_stream->data()), trt_model_stream->size());
+    outfile.write(reinterpret_cast<char *>(plan->data()), plan->size());
     outfile.close();
-    network->destroy();
-    parser->destroy();
-    builder->destroy();
-    config->destroy();
+    if(network) {
+      delete network;
+    }
+    if(parser) {
+      delete parser;
+    }
+    if(builder) {
+      delete builder;
+    }
+    if(config) {
+      delete config;
+    }
+    if(plan) {
+      delete plan;
+    }
   }
   net_ptr_.reset(new Tn::trtNet(engine_file));
 

--- a/perception/lidar_centerpoint/CMakeLists.txt
+++ b/perception/lidar_centerpoint/CMakeLists.txt
@@ -4,9 +4,6 @@ project(lidar_centerpoint)
 find_package(autoware_cmake REQUIRED)
 autoware_package()
 
-# TODO: remove (https://github.com/autowarefoundation/autoware.universe/issues/850)
-add_compile_options(-Wno-deprecated-declarations -Wno-unknown-pragmas)
-
 option(CUDA_VERBOSE "Verbose output of CUDA modules" OFF)
 
 # set flags for CUDA availability
@@ -142,6 +139,13 @@ if(TRT_AVAIL AND CUDA_AVAIL AND CUDNN_AVAIL)
     ${CUDA_curand_LIBRARY}
     ${CUDNN_LIBRARY}
     centerpoint_cuda_libraries
+  )
+
+  # To suppress unknown-pragmas error. The root-cause is CUB library in CUDA 11.6.
+  # This issue was fixed by https://github.com/NVIDIA/cub/commit/7d608bf1dc14553e2fb219eabeed80b76621b6fe
+  target_include_directories(centerpoint
+    SYSTEM PUBLIC
+      ${CUDA_INCLUDE_DIRS}
   )
 
   ## node ##

--- a/perception/lidar_centerpoint/lib/include/tensorrt_wrapper.hpp
+++ b/perception/lidar_centerpoint/lib/include/tensorrt_wrapper.hpp
@@ -29,7 +29,7 @@ struct Deleter
   void operator()(T * obj) const
   {
     if (obj) {
-      obj->destroy();
+      delete obj;
     }
   }
 };
@@ -82,6 +82,7 @@ private:
   bool createContext();
 
   unique_ptr<nvinfer1::IRuntime> runtime_ = nullptr;
+  unique_ptr<nvinfer1::IHostMemory> plan_ = nullptr;
   unique_ptr<nvinfer1::ICudaEngine> engine_ = nullptr;
 };
 

--- a/perception/lidar_centerpoint/lib/src/tensorrt_wrapper.cpp
+++ b/perception/lidar_centerpoint/lib/src/tensorrt_wrapper.cpp
@@ -104,7 +104,13 @@ bool TensorRTWrapper::parseONNX(
 
   std::cout << "Applying optimizations and building TRT CUDA engine (" << onnx_path << ") ..."
             << std::endl;
-  engine_ = unique_ptr<nvinfer1::ICudaEngine>(builder->buildEngineWithConfig(*network, *config));
+  plan_ = unique_ptr<nvinfer1::IHostMemory>(builder->buildSerializedNetwork(*network, *config));
+  if (!plan_) {
+    std::cout << "Fail to create serialized network" << std::endl;
+    return false;
+  }
+  engine_ =
+    unique_ptr<nvinfer1::ICudaEngine>(runtime_->deserializeCudaEngine(plan_->data(), plan_->size()));
   if (!engine_) {
     std::cout << "Fail to create engine" << std::endl;
     return false;
@@ -116,9 +122,8 @@ bool TensorRTWrapper::parseONNX(
 bool TensorRTWrapper::saveEngine(const std::string & engine_path)
 {
   std::cout << "Writing to " << engine_path << std::endl;
-  auto serialized = unique_ptr<nvinfer1::IHostMemory>(engine_->serialize());
   std::ofstream file(engine_path, std::ios::out | std::ios::binary);
-  file.write(reinterpret_cast<const char *>(serialized->data()), serialized->size());
+  file.write(reinterpret_cast<const char *>(plan_->data()), plan_->size());
   return true;
 }
 
@@ -139,8 +144,7 @@ bool TensorRTWrapper::loadEngine(const std::string & engine_path)
   }
 
   std::cout << "Loading from " << engine_path << std::endl;
-  engine_ =
-    unique_ptr<nvinfer1::ICudaEngine>(runtime_->deserializeCudaEngine(buffer.get(), size, nullptr));
+  engine_ = unique_ptr<nvinfer1::ICudaEngine>(runtime_->deserializeCudaEngine(buffer.get(), size));
   return true;
 }
 

--- a/perception/lidar_centerpoint/lib/src/tensorrt_wrapper.cpp
+++ b/perception/lidar_centerpoint/lib/src/tensorrt_wrapper.cpp
@@ -109,8 +109,8 @@ bool TensorRTWrapper::parseONNX(
     std::cout << "Fail to create serialized network" << std::endl;
     return false;
   }
-  engine_ =
-    unique_ptr<nvinfer1::ICudaEngine>(runtime_->deserializeCudaEngine(plan_->data(), plan_->size()));
+  engine_ = unique_ptr<nvinfer1::ICudaEngine>(
+    runtime_->deserializeCudaEngine(plan_->data(), plan_->size()));
   if (!engine_) {
     std::cout << "Fail to create engine" << std::endl;
     return false;

--- a/perception/tensorrt_yolo/CMakeLists.txt
+++ b/perception/tensorrt_yolo/CMakeLists.txt
@@ -4,9 +4,6 @@ project(tensorrt_yolo)
 find_package(autoware_cmake REQUIRED)
 autoware_package()
 
-# TODO: remove (https://github.com/autowarefoundation/autoware.universe/issues/850)
-add_compile_options(-Wno-deprecated-declarations)
-
 option(CUDA_VERBOSE "Verbose output of CUDA modules" OFF)
 
 find_package(OpenCV REQUIRED)

--- a/perception/tensorrt_yolo/lib/include/trt_yolo.hpp
+++ b/perception/tensorrt_yolo/lib/include/trt_yolo.hpp
@@ -59,7 +59,7 @@ struct Deleter
   void operator()(T * obj) const
   {
     if (obj) {
-      obj->destroy();
+      delete obj;
     }
   }
 };
@@ -130,6 +130,7 @@ public:
 
 private:
   unique_ptr<nvinfer1::IRuntime> runtime_ = nullptr;
+  unique_ptr<nvinfer1::IHostMemory> plan_ = nullptr;
   unique_ptr<nvinfer1::ICudaEngine> engine_ = nullptr;
   unique_ptr<nvinfer1::IExecutionContext> context_ = nullptr;
   cudaStream_t stream_ = nullptr;

--- a/perception/tensorrt_yolo/lib/src/trt_yolo.cpp
+++ b/perception/tensorrt_yolo/lib/src/trt_yolo.cpp
@@ -257,8 +257,8 @@ Net::Net(
     std::cout << "Fail to create serialized network" << std::endl;
     return;
   }
-  engine_ =
-    unique_ptr<nvinfer1::ICudaEngine>(runtime_->deserializeCudaEngine(plan_->data(), plan_->size()));
+  engine_ = unique_ptr<nvinfer1::ICudaEngine>(
+    runtime_->deserializeCudaEngine(plan_->data(), plan_->size()));
   if (!prepare()) {
     std::cout << "Fail to create engine" << std::endl;
     return;

--- a/perception/tensorrt_yolo/lib/src/trt_yolo.cpp
+++ b/perception/tensorrt_yolo/lib/src/trt_yolo.cpp
@@ -65,8 +65,7 @@ void Net::load(const std::string & path)
   file.read(buffer, size);
   file.close();
   if (runtime_) {
-    engine_ =
-      unique_ptr<nvinfer1::ICudaEngine>(runtime_->deserializeCudaEngine(buffer, size, nullptr));
+    engine_ = unique_ptr<nvinfer1::ICudaEngine>(runtime_->deserializeCudaEngine(buffer, size));
   }
   delete[] buffer;
 }
@@ -253,9 +252,15 @@ Net::Net(
 
   // Build engine
   std::cout << "Applying optimizations and building TRT CUDA engine..." << std::endl;
-  engine_ = unique_ptr<nvinfer1::ICudaEngine>(builder->buildEngineWithConfig(*network, *config));
+  plan_ = unique_ptr<nvinfer1::IHostMemory>(builder->buildSerializedNetwork(*network, *config));
+  if (!plan_) {
+    std::cout << "Fail to create serialized network" << std::endl;
+    return;
+  }
+  engine_ =
+    unique_ptr<nvinfer1::ICudaEngine>(runtime_->deserializeCudaEngine(plan_->data(), plan_->size()));
   if (!prepare()) {
-    std::cout << "Fail to prepare engine" << std::endl;
+    std::cout << "Fail to create engine" << std::endl;
     return;
   }
 }
@@ -263,9 +268,8 @@ Net::Net(
 void Net::save(const std::string & path) const
 {
   std::cout << "Writing to " << path << "..." << std::endl;
-  auto serialized = unique_ptr<nvinfer1::IHostMemory>(engine_->serialize());
   std::ofstream file(path, std::ios::out | std::ios::binary);
-  file.write(reinterpret_cast<const char *>(serialized->data()), serialized->size());
+  file.write(reinterpret_cast<const char *>(plan_->data()), plan_->size());
 }
 
 void Net::infer(std::vector<void *> & buffers, const int batch_size)

--- a/perception/traffic_light_classifier/CMakeLists.txt
+++ b/perception/traffic_light_classifier/CMakeLists.txt
@@ -4,9 +4,6 @@ project(traffic_light_classifier)
 find_package(autoware_cmake REQUIRED)
 autoware_package()
 
-# TODO: remove (https://github.com/autowarefoundation/autoware.universe/issues/850)
-add_compile_options(-Wno-deprecated-declarations)
-
 option(CUDA_VERBOSE "Verbose output of CUDA modules" OFF)
 # set flags for CUDA availability
 option(CUDA_AVAIL "CUDA available" OFF)

--- a/perception/traffic_light_classifier/utils/trt_common.cpp
+++ b/perception/traffic_light_classifier/utils/trt_common.cpp
@@ -121,7 +121,8 @@ bool TrtCommon::buildEngineFromOnnx(std::string onnx_file_path, std::string outp
   if (!plan) {
     return false;
   }
-  engine_ = UniquePtr<nvinfer1::ICudaEngine>(runtime_->deserializeCudaEngine(plan->data(), plan->size()));
+  engine_ =
+    UniquePtr<nvinfer1::ICudaEngine>(runtime_->deserializeCudaEngine(plan->data(), plan->size()));
   if (!engine_) {
     return false;
   }

--- a/perception/traffic_light_classifier/utils/trt_common.cpp
+++ b/perception/traffic_light_classifier/utils/trt_common.cpp
@@ -45,6 +45,7 @@ TrtCommon::TrtCommon(std::string model_path, std::string precision)
   is_initialized_(false),
   max_batch_size_(1)
 {
+  runtime_ = UniquePtr<nvinfer1::IRuntime>(nvinfer1::createInferRuntime(logger_));
 }
 
 void TrtCommon::setup()
@@ -86,9 +87,8 @@ bool TrtCommon::loadEngine(std::string engine_file_path)
   std::stringstream engine_buffer;
   engine_buffer << engine_file.rdbuf();
   std::string engine_str = engine_buffer.str();
-  runtime_ = UniquePtr<nvinfer1::IRuntime>(nvinfer1::createInferRuntime(logger_));
   engine_ = UniquePtr<nvinfer1::ICudaEngine>(runtime_->deserializeCudaEngine(
-    reinterpret_cast<const void *>(engine_str.data()), engine_str.size(), nullptr));
+    reinterpret_cast<const void *>(engine_str.data()), engine_str.size()));
   return true;
 }
 
@@ -117,19 +117,22 @@ bool TrtCommon::buildEngineFromOnnx(std::string onnx_file_path, std::string outp
     return false;
   }
 
-  engine_ = UniquePtr<nvinfer1::ICudaEngine>(builder->buildEngineWithConfig(*network, *config));
+  auto plan = UniquePtr<nvinfer1::IHostMemory>(builder->buildSerializedNetwork(*network, *config));
+  if (!plan) {
+    return false;
+  }
+  engine_ = UniquePtr<nvinfer1::ICudaEngine>(runtime_->deserializeCudaEngine(plan->data(), plan->size()));
   if (!engine_) {
     return false;
   }
 
   // save engine
-  nvinfer1::IHostMemory * data = engine_->serialize();
   std::ofstream file;
   file.open(output_engine_file_path, std::ios::binary | std::ios::out);
   if (!file.is_open()) {
     return false;
   }
-  file.write((const char *)data->data(), data->size());
+  file.write((const char *)plan->data(), plan->size());
   file.close();
 
   return true;

--- a/perception/traffic_light_ssd_fine_detector/CMakeLists.txt
+++ b/perception/traffic_light_ssd_fine_detector/CMakeLists.txt
@@ -4,9 +4,6 @@ project(traffic_light_ssd_fine_detector)
 find_package(autoware_cmake REQUIRED)
 autoware_package()
 
-# TODO: remove (https://github.com/autowarefoundation/autoware.universe/issues/850)
-add_compile_options(-Wno-deprecated-declarations)
-
 option(CUDA_VERBOSE "Verbose output of CUDA modules" OFF)
 
 find_package(OpenCV REQUIRED)

--- a/perception/traffic_light_ssd_fine_detector/lib/include/trt_ssd.hpp
+++ b/perception/traffic_light_ssd_fine_detector/lib/include/trt_ssd.hpp
@@ -31,7 +31,7 @@ struct Deleter
   void operator()(T * obj) const
   {
     if (obj) {
-      obj->destroy();
+      delete obj;
     }
   }
 };
@@ -87,6 +87,7 @@ public:
 
 private:
   unique_ptr<nvinfer1::IRuntime> runtime_ = nullptr;
+  unique_ptr<nvinfer1::IHostMemory> plan_ = nullptr;
   unique_ptr<nvinfer1::ICudaEngine> engine_ = nullptr;
   unique_ptr<nvinfer1::IExecutionContext> context_ = nullptr;
   cudaStream_t stream_ = nullptr;

--- a/perception/traffic_light_ssd_fine_detector/lib/src/trt_ssd.cpp
+++ b/perception/traffic_light_ssd_fine_detector/lib/src/trt_ssd.cpp
@@ -37,7 +37,7 @@ void Net::load(const std::string & path)
   file.close();
   if (runtime_) {
     engine_ =
-      unique_ptr<nvinfer1::ICudaEngine>(runtime_->deserializeCudaEngine(buffer, size, nullptr));
+      unique_ptr<nvinfer1::ICudaEngine>(runtime_->deserializeCudaEngine(buffer, size));
   }
   delete[] buffer;
 }
@@ -136,7 +136,13 @@ Net::Net(
 
   // Build engine
   std::cout << "Applying optimizations and building TRT CUDA engine..." << std::endl;
-  engine_ = unique_ptr<nvinfer1::ICudaEngine>(builder->buildEngineWithConfig(*network, *config));
+  plan_ = unique_ptr<nvinfer1::IHostMemory>(builder->buildSerializedNetwork(*network, *config));
+  if (!plan_) {
+    std::cout << "Fail to create serialized network" << std::endl;
+    return;
+  }
+  engine_ =
+    unique_ptr<nvinfer1::ICudaEngine>(runtime_->deserializeCudaEngine(plan_->data(), plan_->size()));
   if (!engine_) {
     std::cout << "Fail to create engine" << std::endl;
     return;
@@ -151,9 +157,8 @@ Net::Net(
 void Net::save(const std::string & path)
 {
   std::cout << "Writing to " << path << "..." << std::endl;
-  auto serialized = unique_ptr<nvinfer1::IHostMemory>(engine_->serialize());
   std::ofstream file(path, std::ios::out | std::ios::binary);
-  file.write(reinterpret_cast<const char *>(serialized->data()), serialized->size());
+  file.write(reinterpret_cast<const char *>(plan_->data()), plan_->size());
 }
 
 void Net::infer(std::vector<void *> & buffers, const int batch_size)

--- a/perception/traffic_light_ssd_fine_detector/lib/src/trt_ssd.cpp
+++ b/perception/traffic_light_ssd_fine_detector/lib/src/trt_ssd.cpp
@@ -36,8 +36,7 @@ void Net::load(const std::string & path)
   file.read(buffer, size);
   file.close();
   if (runtime_) {
-    engine_ =
-      unique_ptr<nvinfer1::ICudaEngine>(runtime_->deserializeCudaEngine(buffer, size));
+    engine_ = unique_ptr<nvinfer1::ICudaEngine>(runtime_->deserializeCudaEngine(buffer, size));
   }
   delete[] buffer;
 }
@@ -141,8 +140,8 @@ Net::Net(
     std::cout << "Fail to create serialized network" << std::endl;
     return;
   }
-  engine_ =
-    unique_ptr<nvinfer1::ICudaEngine>(runtime_->deserializeCudaEngine(plan_->data(), plan_->size()));
+  engine_ = unique_ptr<nvinfer1::ICudaEngine>(
+    runtime_->deserializeCudaEngine(plan_->data(), plan_->size()));
   if (!engine_) {
     std::cout << "Fail to create engine" << std::endl;
     return;

